data/tutorials/getting-started/02_fst_program.md

```diff
@@ -0,0 +1,310 @@
+
+To create a module, let's create a file named `lib/hello.ml`:
+```shell
+$ echo 'let world = "Hello from a module"' > lib/hello.ml
+```
+
+Here is a new version of the `bin/main.ml` file and execution of the resulting project:
+```shell
+$ echo 'let () = Printf.printf "%s\n" Hello.world' > bin/main.ml
+
+$ dune exec hello
+Entering directory '/home/cuihtlauac/caml/ocaml.org'
+Hello from a module
+```
+
+The file `lib/hello.ml` contains the module named `Hello`, which in turn defines a string named `world`. This definition is referred to as `Hello.world` from the `bin/main.ml` file.
+
+Dune can launch UTop to access to the modules exposed by a project interactively. Here's how:
+```shell 
+$ dune utop
+```
+
+Then, inside the `utop` toplevel, it is possible to inspect our `Hello` module, just like we did when we examined the `Option` module in the previous tutorial.
+```ocaml
+# #show Hello;;
+module Hello : sig val world : string end
+```
+
+## Defining Module Interfaces
+
+What UTop's `#show` command displays is an API, the list of definitions provided by a module, which is called a _module interface_. In a similar way, module can be defined by `.ml` files. It is also possible to define module interfaces from files. Such files have a `.mli` extension and must share the same name part. 
+```shell
+$ echo 'val world : string' > lib/hello.mli
+```
+
+Observe that only what is between `sig` and `end` needs to written in the interface file. This is explained in the tutorial dedicated to [modules](link_modules).
+
+Module interfaces are also used to create _private_ definitions. A module definition is private if it is not listed in its corresponding interface. If no interface file exists, everyting is public.
+
+Edit the `lib/hello.ml` file to add the `mundo` definition, as shown below:
+```ocaml
+let mundo = "¡Hola Mundo!"
+let world = mundo
+```
+
+Also edit the `bin/main.ml` file like this:
+```ocaml 
+let () = Printf.printf "%s\n" Hello.mundo
+```
+
+Trying to compile this fails.
+```shell
+$ dune build
+Entering directory '/home/cuihtlauac/caml/ocaml.org'
+File "hello/bin/main.ml", line 1, characters 30-41:
+1 | let () = Printf.printf "%s\n" Hello.mundo
+                                  ^^^^^^^^^^^
+Error: Unbound value Hello.mundo
+```
+
+This is because we haven't changed `hello.mli`. Since it does not list `mundo`, it is therefore private.
+
+## Installing and Using Modules from a Package
+
+OCaml has an active community of open-source contributors. Most projects are avaiable using the opam package manager, which you installed in the "Install OCaml" tutorial. The following section shows how to install and use a package from opam's open-source repository.
+
+To illustrate this, let's turn our modest `hello` project into a web server using Anton Bachin's Dream web framework. First install the `dream` package with this command:
+```shell
+$ opam install dream
```

Comment:
I get so many [y/n] questions and some errors about not having x installed... should we speak to that or keep it simple? I'll send a screenshot in Slack.